CHANGELOG.md

```diff
@@ -1,5 +1,344 @@
+  [pull/369]: https://github.com/cloudflare/wrangler/pull/369
+
+  - **feat: unify upload form building - [ashleymichal], [pull/329]**
+
+  This PR brings consistency to the way that metadata is handled during Wrangler's `preview` and `build` commands: previously, a `metadata.json` file handled the bindings for your Wrangler project, but because it was handled differently for various project types (such as `webpack` or `rust`), it led to inconsistent behavior during the upload process. This PR removes usage of `metadata.json` in favor of building metadata for each project type in-memory, which will then be uploaded directly to Workers platform. This work is foundational for improved Workers KV support in Wrangler 🙌
+
+  [ashleymichal]: https://github.com/ashleymichal
+  [pull/329]: https://github.com/cloudflare/wrangler/pull/329
+
+  - **Use stdin instead of arguments for wrangler config - [xtuc], [pull/329]**
+
+  We've made the `wrangler config` command interactive – the previous version of the command, `wrangler config $email $apiKey`, would be captured by your terminal's history, often exposing that information in a `~/.bash_history` or a similar file. The new version of `wrangler config` will prompt you for your `email` and `api_key` via `stdin`.
+
+  In addition, this PR also adds support for a `WRANGLER_HOME` environment variable, which will be the location for Wrangler's "home" directory, if you need to customize where Wrangler saves its configuration information.
+
+  [xtuc]: https://github.com/xtuc
+  [pull/329]: https://github.com/cloudflare/wrangler/pull/239
+
+  - **Settings: make global config file optional - [AaronO], [pull/225]**
+
+  Previously, Wrangler required a global configuration file to be able to run. As many users may use Wrangler in situations where they don't have an interactive terminal, meaning they can't instantiate a config file using `wrangler config`, this PR allows Wrangler to run even if the config file doesn't exist. This change means that users can also configure Wrangler exclusively with environment variables, using `$CF_API_KEY` for your Cloudflare API key, and `$CF_EMAIL` for your Cloudflare account email.
+
+  [AaronO]: https://github.com/AaronO
+  [pull/225]: https://github.com/cloudflare/wrangler/pull/225
+
+  - **Don't break `wrangler preview` when kv namespaces present - [ashleymichal], [pull/353]**
+
+  This PR fixes a critical bug where using `wrangler preview` on a project with [Workers KV][kv] namespaces causes the command to throw an error.
+
+  [ashleymichal]: https://github.com/ashleymichal
+  [pull/353]: https://github.com/cloudflare/wrangler/pull/353
+
+  - **feat kv config - [ashleymichal], [pull/333]**
+
+  If you've tried to use [Workers KV][kv] in Wrangler, you've probably had a bad time! This PR gets us closer to a first-class KV experience inside of Wrangler, by beginning to support KV namespaces in Wrangler's configuration process. TODO unclear what functional changes
```

Comment:
Note: This PR just implements "reading the KV binding name and namespace_id for each namespace, and uploading to the API." not sure the best way to phrase that

Reply:
err thank you, need to come back and implement this. i talked to ashley and understand what this does